File: LeetCode/cpp/base_7.cpp
#include <string>
#include <math.h>
#include <iostream>

class Solution {
public:
    std::string convertToBase7(int num) {
        int number = abs(num);
        int base_7_form_of_number = 0;

        for (int i = 0; number != 0; number /= 7, i++)
        {
            int temp = number % 7;
            base_7_form_of_number += temp * pow(10, i);
        }

        std::string answer = abs(num) == num ? std::to_string(base_7_form_of_number) : '-' + std::to_string(base_7_form_of_number);
        return answer;
    }
};

int main () {
    Solution Jokerge;
    std::cout << Jokerge.convertToBase7(-7);
    std::cout << Jokerge.convertToBase7(-7);
    std::cout << Jokerge.convertToBase7(100);
}

File: LeetCode/go/leetcode/base_7.go
package leetcode

import (
	"math"
	"strconv"
)

func convertToBase7(num int) string {
	var (
		abs_num int = int(math.Abs(float64(num)));
		base_7_number int;
		is_negative bool = abs_num != num;
		answer string;
	)

	for i := 0; abs_num != 0; i++{
		base_7_number += (abs_num % 7) * int(math.Pow(float64(10), float64(i)));
		abs_num /= 7;
	}

	if is_negative {
		answer = "-" + 	strconv.Itoa(base_7_number);
	} else {
		answer = strconv.Itoa(base_7_number);
	}

	return answer
}